### DISCUSSION TITLE
Function sum

### DISCUSSION THREAD

**Discussion by BYS543** [yalmip/YALMIP#958] [General discussion] [answered]
Hi, Johan,
    recently, when I use "sum", i feel some confusion,
    in matlab grammar, if i run
```
   A=[1,2,3;4,5,6;7,8,9;];
   B=[1,3];
   sum(A(B,:))
```
then result shows
`ans = [8,10,12]`
which is the sum of row 1 and 3
however in  Yalmip
if i define
```
A = sdpvar(3,3,'full');
B = [1,3];
Con = [sum(A(B,:)) ==0];
```
it might not be equal to(i think,because i test it)
```
A = sdpvar(3,3,'full');
B = [1,3];
Con = [ones(1,2)*A(B,:) ==0];
```

**Comment by johanlofberg** [comment, marked as answer]
The operator returns the correct thing here

```
>> sum(A(B,:)) - (A(1,:) + A(3,:))

ans =

     0     0     0
```

as seen by sdisplay

```
>> sdisplay(sum(A(B,:)))

ans =

  1×3 cell array

    {'A(1,1)+A(3,1)'}    {'A(1,2)+A(3,2)'}    {'A(1,3)+A(3,3)'}
```

and the two versions you show are the same thing

```
>> sum(A(B,:))-ones(1,2)*A(B,:)

ans =

     0     0     0
```

**Comment by BYS543** [comment]
Thank you, Johan. can `sdisplay`  to display a constraint(i find the type is `lmi`)? Such as "x + y <=1", or do other functions  have the function? By the way, if i define wt as a array of `wt = sdpvar(42,20)`, if  i use `sum(wt([1,2],:))`, it shows as,
` 'wt(1)+wt(2)'    'wt(43)+wt(44)'    'wt(85)+wt(86)'` instead a two dimensional array just like `wt(1,2)`.
if the version of Yalimip is not correct?

**Reply by johanlofberg** [reply]
sum(wt([1,2],:)) is a vector or size 1x20 and the display you show is the correct. Linear indexing is used (sdisplay is not even guaranteed to show the right name, YALMIP has no notion of variable names, the fact that sdisplay here is capable of writing wt(42) is due to some hacks. It could have said internalvariable(1765) or similiar)

To sdisplay a constraint, cast as sdpvar, sdisplay(sdpvar(x + y <= 1)), and the normalized expression in the constraint is shown

**Reply by BYS543** [reply]
i understand, thank you, Johan.
